docs/apache-airflow/security/index.rst

```diff
@@ -0,0 +1,165 @@
+   code on workers by configuring connections and need to be trusted not
+   to abuse these privileges. They have access to sensitive credentials
+   and can modify them. By default, they don't have access to
+   system-level configuration. They should be trusted not to misuse
+   sensitive information accessible through connection configuration.
+   They also have the ability to create a Webserver Denial of Service
+   situation and should be trusted not to misuse this capability.
+
+2. **Connection configuration users**: They configure connections and
+   potentially execute code on workers during DAG execution. Trust is
+   required to prevent misuse of these privileges. They have full access
+   to sensitive credentials stored in connections and can modify them.
+   Access to sensitive information through connection configuration
+   should be trusted not to be abused. They also have the ability to
+   create a Webserver Denial of Service situation and should be trusted
+   not to misuse this capability.
+
+3. **Operations users**: They have access to DAG execution status via
+   the UI. Currently, Airflow lacks full protection for accessing groups
+   of DAGs' history and execution. They can perform actions such as
+   clearing, re-running, triggering DAGs, and changing parameters.
+   Depending on access restrictions, they may also have access to
+   editing variables and viewing Airflow configuration. They should not
+   have access to sensitive system-level information or connections, and
+   they should not be able to access sensitive task information unless
+   deliberately exposed in logs by DAG authors. They should be trusted
+   not to abuse their privileges, as they can potentially overload the
+   server and cause Denial of Service situations.
+
+Responsibilities of Deployment Managers
+---------------------------------------
+
+Deployment Managers determine access levels and understand the potential
+damage users can cause. Some Deployment Managers may further limit
+access through fine-grained privileges for the **Authenticated UI
+users**. However, these limitations are outside the basic Airflow's
+security model and are at the discretion of Deployment Managers.
+Examples of fine-grained access control include (but are not limited
+to):
+
+-  Limiting login permissions: Restricting the accounts that users can
+   log in with, allowing only specific accounts or roles belonging to
+   access the Airflow system.
+
+-  Access restrictions to views or DAGs: Controlling user access to
+   certain views or specific DAGs, ensuring that users can only view or
+   interact with authorized components.
+
+-  Implementing static code analysis and code review: Introducing
+   processes such as static code analysis and code review as part of the
+   DAG submission pipeline. This helps enforce code quality standards,
+   security checks, and adherence to best practices before DAGs are
+   deployed.
```

Comment:
Perhaps out of scope for a document focused on Airflow itself? I don't feel strongly though, happy to keep it.

Reply:
I think it adds value. I think the main point I have here is that Airlfow is "inherently insecure" because DAG authors can do everything. And this is both yes and no. It's kind of symilar to code submission to Software Repository - anyone can submit a code, but we have a gate in form of review of the code before it is submitted and it's better than anything you could do automatically I think (if you treat it seriously).

It is part of the general lack of understanding that you can apply standard engineering practices to your DAG pipeline and that can bring one level of security that might be "good enough" for many of our users.  And it also provides extensibility of the security model - as for example one could have already a very simple multi-tenanti implementation of Airlfow if they add code review and automated CI checks to the mix of things. 

I had this many times when I talked to the users - and when they already started to doubt into their "split teams"  idea , I thrown that one in the mix and they were " ah yaeah, indeed that might work".